src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -881,6 +881,20 @@ class CutLMHead : public ov::pass::MatcherPass {
             auto matched_matmul = std::static_pointer_cast<ov::op::v0::MatMul>(matched_node_matmul);
             auto matched_result = std::static_pointer_cast<ov::op::v0::Result>(matched_node_result);
 
+            // Some LLMs add intermediate hidden state outputs that can interfere with LM head detection.
+            // Use "hidden_output_name" rt_info to distinguish them from the actual logits output.
+            // For example, Eagle-3 target/draft models add "last_hidden_state" output which should be skipped.
+            if (model_rt_info.count(ov::npuw::LLMCompiledModel::hidden_output_name_key)) {
+                const auto& hidden_output_name =
+                    model_rt_info.at(ov::npuw::LLMCompiledModel::hidden_output_name_key).as<std::string>();
+                const auto& result_output_names = matched_result->output(0).get_names();
+                for (const auto& name : result_output_names) {
+                    if (name == hidden_output_name) {
+                        return false;
+                    }
+                }
```

Comment:
So instead of this you can just check matched_result's RT_INFO to check if it was added manually, and skip the cut if that's the case.